downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/GenerateGOAnnotationFile.java

```diff
@@ -0,0 +1,413 @@
+    private static Integer assignDateForGOALine(GKInstance entityInst, String goaLine) throws Exception {
+        int instanceDate;
+        Collection<GKInstance> modifiedInstances = entityInst.getAttributeValuesList(ReactomeJavaConstants.modified);
+        if (modifiedInstances.size() > 0) {
+            List<GKInstance> modifiedInstancesList = new ArrayList<>(modifiedInstances);
+            GKInstance mostRecentModifiedInst = modifiedInstancesList.get(modifiedInstancesList.size() - 1);
+            instanceDate = Integer.valueOf(mostRecentModifiedInst.getAttributeValue(ReactomeJavaConstants.dateTime).toString().split(" ")[0].replaceAll("-", ""));
+        } else {
+            GKInstance createdInst = (GKInstance) entityInst.getAttributeValue(ReactomeJavaConstants.created);
+            instanceDate = Integer.valueOf(createdInst.getAttributeValue(ReactomeJavaConstants.dateTime).toString().split(" ")[0].replaceAll("-", ""));
+        }
+
+        // Stores date in global hash that allows date value to be updated if a more recent date was found.
+        if (dates.get(goaLine) == null) {
+            dates.put(goaLine, instanceDate);
+        } else {
+            if (instanceDate > dates.get(goaLine)) {
+                dates.put(goaLine, instanceDate);
+            }
+        }
+        return instanceDate;
+    }
+
+    // Generates the third column of the GOA line by checking attributes of the referenceEntity instance.
+    private static String getSecondaryIdentifier(GKInstance referenceEntityInst) throws Exception {
+        if (referenceEntityInst.getAttributeValue(ReactomeJavaConstants.secondaryIdentifier) != null) {
+            return referenceEntityInst.getAttributeValue(ReactomeJavaConstants.secondaryIdentifier).toString();
+        } else if (referenceEntityInst.getAttributeValue(ReactomeJavaConstants.geneName) != null) {
+            return referenceEntityInst.getAttributeValue(ReactomeJavaConstants.geneName).toString();
+        } else {
+            return referenceEntityInst.getAttributeValue(ReactomeJavaConstants.identifier).toString();
+        }
+    }
+
+    // Checks if this PhysicalEntity is one that is made of multiple PhysicalEntity instances.
+    private static boolean multiInstancePhysicalEntity(SchemaClass physicalEntitySchemaClass) {
+        if (physicalEntitySchemaClass.isa(ReactomeJavaConstants.Complex) || physicalEntitySchemaClass.isa(ReactomeJavaConstants.EntitySet) || physicalEntitySchemaClass.isa(ReactomeJavaConstants.Polymer)) {
+            return true;
+        }
+        return false;
+    }
+
+    // Returns all constituent PhysicalEntity instances of a Multi-instance PhysicalEntity
+    private static Set<GKInstance> getMultiInstanceSubInstances(GKInstance physicalEntityInst) throws Exception {
+        SchemaClass physicalEntitySchemaClass = physicalEntityInst.getSchemClass();
+        String subunitType = null;
```

Comment:
A separate method to find the `subunitType` would be useful here.  You could use it in a pipeline like this:
```
Collection<GKInstance> subunitInstances =
	physicalEntityInst.getAttributeValuesList(getSubunitType(physicalEntityInst));

return subunitInstances
	.stream()
	.flatMap(subunitInst -> getBiologicalProcessProteins(subunitInst).stream())
	.collect(Collectors.toSet());
```

Reply:
I created the `getSubunitType` method, but the return statement did not work -- intellij forced a refactor into this: 
```        
return subunitInstances
                .stream()
                .flatMap(BiologicalProcessAnnotationBuilder::apply)
                .collect(Collectors.toSet());
```
This still caused a build problem, so for now I've kept the for loop

Reply:
Could we look at that together?  Maybe IntelliJ made a wrong assumption about the refactor.